Makefile

```diff
@@ -12,130 +12,222 @@
 build-linux:
-	GOOS=linux GOARCH=$(ARCH) CGO_ENABLED=0 go build -o aws-k8s-agent -ldflags "-s -w $(LDFLAGS)" ./cmd/aws-k8s-agent/
-	GOOS=linux GOARCH=$(ARCH) CGO_ENABLED=0 go build -o aws-cni -ldflags " -s -w $(LDFLAGS)" ./cmd/routed-eni-cni-plugin/
-	GOOS=linux GOARCH=$(ARCH) CGO_ENABLED=0 go build -o grpc-health-probe -ldflags "-s -w $(LDFLAGS)" ./cmd/grpc-health-probe/
-
-# Download portmap plugin
-download-portmap:
-	mkdir -p tmp/downloads
-	mkdir -p tmp/plugins
-	curl -L -o tmp/downloads/cni-plugins-$(ARCH).tgz https://github.com/containernetworking/plugins/releases/download/v0.7.5/cni-plugins-$(ARCH)-v0.7.5.tgz
-	tar -vxf tmp/downloads/cni-plugins-$(ARCH).tgz -C tmp/plugins
-	cp tmp/plugins/portmap .
-	rm -rf tmp
-
-# Build CNI Docker image
+	go build $(BUILD_FLAGS) -o aws-k8s-agent     ./cmd/aws-k8s-agent
+	go build $(BUILD_FLAGS) -o aws-cni           ./cmd/routed-eni-cni-plugin
+	go build $(BUILD_FLAGS) -o grpc-health-probe ./cmd/grpc-health-probe
+
+# Build VPC CNI plugin & agent container image.
 docker:
-	@docker build $(DOCKER_ARGS) --build-arg arch="$(ARCH)" -f scripts/dockerfiles/Dockerfile.release -t "$(IMAGE):$(VERSION)" .
-	@echo "Built Docker image \"$(IMAGE):$(VERSION)\""
+	docker build $(DOCKER_BUILD_FLAGS) \
+		-f scripts/dockerfiles/Dockerfile.release \
+		-t "$(IMAGE_NAME)" \
+		.
+	@echo "Built Docker image \"$(IMAGE_NAME)\""
 
+# Run the built cni container image to use in functional testing
 docker-func-test: docker
-	docker run $(DOCKER_ARGS) -it "$(IMAGE):$(VERSION)"
+	docker run $(DOCKER_RUN_FLAGS) \
+		"$(IMAGE_NAME)"
 
-# unit-test
+# Run unit tests
 unit-test:
-	GOOS=linux CGO_ENABLED=1 go test -v -cover $(ALLPKGS)
+	go test -mod=readonly -v -cover $(ALLPKGS)
 
-# unit-test-race
+# Run unit tests with race detection (can only be run natively)
+unit-test-race: CGO_ENABLED=1
+unit-test-race: GOARCH=
 unit-test-race:
-	GOOS=linux CGO_ENABLED=1 go test -v -cover -race -timeout 10s ./cmd/*/...
-	GOOS=linux CGO_ENABLED=1 go test -v -cover -race -timeout 150s ./pkg/awsutils/...
-	GOOS=linux CGO_ENABLED=1 go test -v -cover -race -timeout 10s ./pkg/k8sapi/...
-	GOOS=linux CGO_ENABLED=1 go test -v -cover -race -timeout 10s ./pkg/networkutils/...
-	GOOS=linux CGO_ENABLED=1 go test -v -cover -race -timeout 10s ./pkg/utils/...
-	GOOS=linux CGO_ENABLED=1 go test -v -cover -race -timeout 10s ./pkg/eniconfig/...
-	GOOS=linux CGO_ENABLED=1 go test -v -cover -race -timeout 10s ./pkg/ipamd/...
-
+	go test -v -mod=readonly -cover -race -timeout 10s  ./cmd/...
+	go test -v -mod=readonly -cover -race -timeout 150s ./pkg/awsutils/...
+	go test -v -mod=readonly -cover -race -timeout 10s  ./pkg/k8sapi/...
+	go test -v -mod=readonly -cover -race -timeout 10s  ./pkg/networkutils/...
+	go test -v -mod=readonly -cover -race -timeout 10s  ./pkg/utils/...
+	go test -v -mod=readonly -cover -race -timeout 10s  ./pkg/eniconfig/...
+	go test -v -mod=readonly -cover -race -timeout 10s  ./pkg/ipamd/...
```

Comment:
Might clean this up in a future commit with:

```Makefile
unit-test-race: COVER_ARGS = -v -mod=readonly -cover -race
    go test $(COVER_ARGS) -timeout 10s ./cmd/*/...
...
```

Reply:
Is this an ask? Or a note for another time? Happy to make the change now if that's welcome.

Reply:
it's a note for a future PR :) no need to make the change here.